app/assets/javascripts/discourse/widgets/default-notification-item.js.es6

```diff
@@ -154,6 +155,17 @@ export const DefaultNotificationItem = createWidget(
           );
         }
       });
+    },
+
+    mouseUp(e) {
+      // dismiss notification on middle click
+      if (e.which === 2 && !this.attrs.read) {
```

Comment:
do we need to look at `e.button` as well? https://stackoverflow.com/a/21224428/17174

Reply:
I believe `e.which` is the "normalized" version of `e.button`. `e.button` can be either `2` or `4` for middle clicks, while `e.which` is always `2`. This felt like a more robust solution.